oauthenticator/github.py

```diff
@@ -109,20 +172,50 @@ def authenticate(self, handler, data=None):
         resp = yield http_client.fetch(req)
         resp_json = json.loads(resp.body.decode('utf8', 'replace'))
 
-        username =  resp_json["login"]
+        username = resp_json["login"]
 
         # Check if user is a member of any whitelisted organizations.
         # This check is performed here, as it requires `access_token`.
         if self.github_organization_whitelist:
             for org in self.github_organization_whitelist:
                 user_in_org = yield self._check_organization_whitelist(org, username, access_token)
-                if user_in_org:
-                    return username
-            else:  # User not found in member list for any organisation
-                return None
-        else:  # no organization whitelisting
-            return username
-
+                if not user_in_org:
+                    # User not found in member list for any organisation
+                    return None
+        # username is now the GitHub userid.
+        if not username:
+            return None
+        userdict = {"name": username}
+        # Now we set up auth_state
+        auth_state = {}
+        auth_state["username"] = username
+        # We may want to do user provisioning in the server container.
+        #  This next bit is about that.
+        #  1) make the username look Unixy
+        #  2) use the GitHub ID as the uid
+        #  3) set list of orgs/gids
+        #  4) set up name/email for .gitconfig
+        safe_chars = set(string.ascii_lowercase + string.digits)
+        safe_username = ''.join(
```

Comment:
Yes, there's `Authenticator.normalize_username`, which does this already on the name returned from this.